main/test/ca/uwaterloo/flix/language/phase/TestNamer.scala

```diff
@@ -453,4 +453,40 @@ class TestNamer extends FunSuite with TestUtils {
     expectError[NameError.UndefinedTypeVar](result)
   }
 
+  test("MismatchedTypeParamKind.01") {
+    val input = "def f(g: Int -> o & o): Int = 123"
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.MismatchedTypeParamKinds](result)
+  }
+
+  test("MismatchedTypeParamKind.02") {
+    val input = "def f(g: Int -> Int & e): e = g(123)"
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.MismatchedTypeParamKinds](result)
+  }
+
+  test("MismatchedTypeParamKind.03") {
+    val input = "def f(s: #{| a}, r: {| a}): Int = 123"
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.MismatchedTypeParamKinds](result)
+  }
+
+  test("MismatchedTypeParamKind.04") {
+    val input = "def f(s: #{X(Int) | a}, r: {x: Int | a}): Int = 123"
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.MismatchedTypeParamKinds](result)
+  }
+
+  test("MismatchedTypeParamKind.05") {
+    val input = "def f(r: {| a}, t: a): Int = 123"
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.MismatchedTypeParamKinds](result)
+  }
+
+  test("MismatchedTypeParamKind.06") {
+    val input = "def f(s: #{| a}, t: a): Int = 123"
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.MismatchedTypeParamKinds](result)
+  }
+
```

Comment:
Could you add a few more test cases where the "fault" occurs inside some type. For example:

```
def f(s: Option[{|a}]): a = 123
```

Just to ensure that we are checking these things correctly in nested types.